packages/platforms/quanta/x86-64/x86-64-quanta-ix1b-rglbmc/onlp/builds/src/x86_64_quanta_ix1b_rglbmc/module/src/thermali.c

```diff
@@ -0,0 +1,33 @@
+/************************************************************
+ * <bsn.cl fy=2014 v=onl>
+ *
+ *           Copyright 2014 Big Switch Networks, Inc.
+ *
+ * Licensed under the Eclipse Public License, Version 1.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *        http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *
+ * </bsn.cl>
+ ************************************************************
+ *
+ *
+ *
+ ***********************************************************/
+#include <onlp/platformi/thermali.h>
+#include "x86_64_quanta_ix1b_rglbmc_log.h"
+
+int
+onlp_thermali_init(void)
```

Comment:
@jnealtowns We will remove all AIM_LOG_MSG statements in LY4R and IX1B.

We think currently BMC support IPMI standard function. If ONL install ipmitool and then it can retrieve information like as fan, thermal, rpsu.

"ipmitool" shall be included into pre-built toolkits.

Actually, IPMI is a well-known industrial standard.
Reading fan, temperature, and rpsu information through onlp may be something like “reinvent the wheel”.

Below message is the example that command line execute "ipmitool sensor":

[root@LY4R-DIAG ~]# ipmitool sensor
Volt_P1V07       | 1.057      | Volts      | ok    | na        | 0.945     | na        | na        | 1.155     | na
Volt_P3V_BAT     | 3.014      | Volts      | ok    | na        | 2.499     | na        | na        | 3.602     | na
Volt_P1V15_STBY  | 1.134      | Volts      | ok    | na        | 1.036     | na        | na        | 1.260     | na
Volt_P3V3_STBY   | 3.325      | Volts      | ok    | na        | 2.975     | na        | na        | 3.623     | na
Volt_P3V3_SUS    | 3.308      | Volts      | ok    | na        | 2.975     | na        | na        | 3.623     | na
Volt_P5V_STBY    | 4.992      | Volts      | ok    | na        | 4.498     | na        | na        | 5.486     | na
Volt_PVDDRAB     | 1.337      | Volts      | ok    | na        | 1.211     | na        | na        | 1.484     | na
Volt_P12V        | 12.222     | Volts      | ok    | na        | 10.773    | na        | na        | 13.167    | na
Temp_Ambient_0   | 27.000     | degrees C  | ok    | na        | na        | na        | 52.000    | 54.000    | 57.000
Temp_Ambient_1   | 35.000     | degrees C  | ok    | na        | na        | na        | 75.000    | 77.000    | 80.000
Temp_Ambient_2   | 36.000     | degrees C  | ok    | na        | na        | na        | 80.000    | 82.000    | 85.000
Temp_Ambient_3   | 30.000     | degrees C  | ok    | na        | na        | na        | 88.000    | 91.000    | 93.000
Temp_Ambient_4   | 34.000     | degrees C  | ok    | na        | na        | na        | 70.000    | 72.000    | 75.000
PCH Thermal Trip | 0x0        | discrete   | 0x0080| na        | na        | na        | na        | na        | na
Volt_CPU_P3V3    | 3.325      | Volts      | ok    | na        | 2.967     | na        | na        | 3.635     | na
Volt_CPU_P1V35   | 1.352      | Volts      | ok    | na        | 1.235     | na        | na        | 1.460     | na
Volt_P3V3        | 3.309      | Volts      | ok    | na        | 2.967     | na        | na        | 3.635     | na
Volt_CPU_P1V8    | 1.784      | Volts      | ok    | na        | 1.627     | na        | na        | 1.980     | na
Volt_CPU_P1V0    | 1.051      | Volts      | ok    | na        | 0.906     | na        | na        | 1.092     | na
Volt_P2V5_STBY   | 2.493      | Volts      | ok    | na        | 2.226     | na        | na        | 2.723     | na
Volt_P1V2_STBY   | 1.205      | Volts      | ok    | na        | 1.088     | na        | na        | 1.313     | na
Volt_P3V3_SFP    | na         | Volts      | na    | na        | 2.967     | na        | na        | 3.635     | na
Volt_P1V0        | 1.051      | Volts      | ok    | na        | 0.906     | na        | na        | 1.092     | na
Event Log        | 0x0        | discrete   | 0x2080| na        | na        | na        | na        | na        | na
Volt_BMC_P3V_BAT | 2.992      | Volts      | ok    | na        | 2.496     | na        | na        | 3.596     | na
Volt_P1V0_SUS    | 0.994      | Volts      | ok    | na        | 0.896     | na        | na        | 1.099     | na
Volt_P1V5_SUS    | 1.477      | Volts      | ok    | na        | 1.344     | na        | na        | 1.645     | na
Volt_PV_DDR_VTT  | 0.658      | Volts      | ok    | na        | 0.595     | na        | na        | 0.728     | na
Volt_PV_DDR_VERF | 0.630      | Volts      | ok    | na        | 0.595     | na        | na        | 0.728     | na
Volt_PV_VCCP     | 0.770      | Volts      | ok    | na        | 0.497     | na        | na        | 1.099     | na
Volt_PV_VNN      | 0.966      | Volts      | ok    | na        | 0.896     | na        | na        | 1.099     | na
Fan_SYS_0        | 1500.000   | RPM        | ok    | na        | 500.000   | 1000.000  | na        | na        | na
Fan_SYS_1        | 1500.000   | RPM        | ok    | na        | 500.000   | 1000.000  | na        | na        | na
Fan_SYS_2        | 1500.000   | RPM        | ok    | na        | 500.000   | 1000.000  | na        | na        | na
Temp_CPU         | 32.000     | degrees C  | ok    | na        | na        | na        | 97.000    | 98.000    | na
Volt_P1V8_SUS    | 1.767      | Volts      | ok    | na        | 1.617     | na        | na        | 1.974     | na
Volt_P1V8_STBY   | 1.767      | Volts      | ok    | na        | 1.617     | na        | na        | 1.974     | na
PSU1_VOLTAGE_OUT | na         | Volts      | na    | na        | 10.800    | 11.400    | 12.600    | 13.560    | na
PSU2_VOLTAGE_OUT | 12.240     | Volts      | ok    | na        | 10.800    | 11.400    | 12.600    | 13.560    | na
PSU1_CURRENT_OUT | na         | Amps       | na    | na        | na        | na        | 10.800    | 13.500    | na
PSU2_CURRENT_OUT | 1.560      | Amps       | ok    | na        | na        | na        | 10.800    | 13.500    | na
PSU1_POWER_OUT   | na         | Watts      | na    | na        | na        | na        | 115.000   | 120.000   | na
PSU2_POWER_OUT   | 18.000     | Watts      | ok    | na        | na        | na        | 115.000   | 120.000   | na
PSU1_TEMP1       | na         | degrees C  | na    | na        | na        | na        | 95.000    | 100.000   | na
PSU1_TEMP2       | na         | degrees C  | na    | na        | na        | na        | 105.000   | 110.000   | na
PSU2_TEMP1       | 34.000     | degrees C  | ok    | na        | na        | na        | 95.000    | 100.000   | na
PSU2_TEMP2       | 29.000     | degrees C  | ok    | na        | na        | na        | 105.000   | 110.000   | na
[root@LY4R-DIAG ~]#